### PR TITLE
Add $XDG_DATA_DIRS checks.

### PR DESCRIPTION
**Pull request by DoomHammer**
As discussed in #899

### CODE REVIEW

**Comment by sjackman** [comment]
Thanks, Piotr. Does this issue only affect `gir`? I would have thought that it would affect any FreeDesktop application that expected to find its data in `XDG_DATA_DIRS`.

**Comment by DoomHammer** [comment]
Problem is I don't know how many of them actually make use of this variable. Based on that I assumed this is the only one until proven otherwise.

**Comment by sjackman** [comment]
I'd suggest casting the net a little wider. If the environment variable `XDG_DATA_DIRS` is declared, it should include `$HOMEBREW_PREFIX/share`. If it's not declared, everything is fine.

**Comment by DoomHammer** [comment]
Fine for me, I will update the PR.

File: Library/Homebrew/diagnostic.rb [hunk 1]
```diff
@@ -755,6 +755,25 @@ def check_user_path_3
...
+        end
+        unless homebrew_in_xdg_data_dirs
+          <<-EOS.undent
+          Homebrew's share was not found in your XDG_DATA_DIRS but you have
+          this variable set to include other locations.
+          Some programs like `vapigen` may not work correctly.
+          Consider setting the XDG_DATA_DIRS for example like so
+              echo 'export XDG_DATA_DIRS="#{share}:$XDG_DATA_DIRS"' >> #{shell_profile}
+        EOS
+        end
```
Comment (sjackman) [inline Library/Homebrew/diagnostic.rb]: Just for style, try (untested)
```ruby
if ENV.key?("XDG_DATA_DIRS") &&
    !ENV["XDG_DATA_DIRS"].split(File::PATH_SEPARATOR).include?(HOMEBREW_PREFIX/"share")
  "..."
end
```

Comment (DoomHammer) [inline Library/Homebrew/diagnostic.rb]: Not so much ;)

Tried to do it in a more compact way, but I think logic won and I lost...

**Comment by sjackman** [comment]
`brew tests` failed: https://travis-ci.org/Linuxbrew/linuxbrew/builds/120258291#L255
I'm getting ready to look at the grand split into `brew` and `homebrew-core`. After fixing the code style and the test, are you ready to merge this PR?

**Comment by DoomHammer** [comment]
Superseded by https://github.com/Linuxbrew/brew/pull/93

**Pull request closed without merging**